### PR TITLE
Add support for filesystem resource 

### PR DESCRIPTION
**Pull request by diogopms**
Add support for filesystem resource
Tests:

- [x] Create a filesystem using Terraform
- [x] Delete the filesystem using terraform
- [x] Updating filesystem (name or description using terraform) 

Client test used:

```terraform {
  required_providers {
    genesiscloud = {
      source  = "genesiscloud/genesiscloud"
      version = "0.0.1-local"
    }
  }
}
provider "genesiscloud" {
  token = "API_KEY"
}

locals {
  region = "NORD-NO-KRS-1"
}

resource "genesiscloud_filesystem" "filesystem" {
  name   = "filesystem-name"
  region = local.region
  size   = 12
  type   = "vast"
}
`

### CODE REVIEW

File: internal/provider/filesystem_resource.go [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+	if diag != nil {
+		resp.Diagnostics.Append(diag...)
+		return
+	}
+	defer cancel()
+
+	body := genesiscloud.UpdateFilesystemJSONRequestBody{}
+
+	body.Name = pointer(data.Name.ValueString())
+	body.Description = pointer(data.Description.ValueString())
```
Comment (skirsten) [inline internal/provider/filesystem_resource.go:307]: We also support updating the size

File: internal/provider/filesystem_types.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	data.CreatedAt = types.StringValue(filesystem.CreatedAt.Format(time.RFC3339))
+	data.Description = types.StringValue(filesystem.Description)
+	data.Id = types.StringValue(filesystem.Id)
+	data.Name = types.StringValue(filesystem.Name)
+
+	if filesystem.MountBasePath != nil {
+		data.MountBasePath = types.StringValue(*filesystem.MountBasePath)
+	} else {
+		data.MountBasePath = types.StringValue("")
+	}
```
Comment (skirsten) [inline internal/provider/filesystem_types.go:62]: Should be null, not empty string.

File: internal/provider/filesystem_types.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	data.MountEndpointRange, diag = types.ListValueFrom(ctx, types.StringType, filesystem.MountEndpointRange)
```
Comment (skirsten) [inline internal/provider/filesystem_types.go:64]: Missing handling of diag

File: internal/provider/filesystem_resource.go [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(), // immutable
+				},
+			}),
+			"mount_endpoint_range": resourceenhancer.Attribute(ctx, schema.ListAttribute{
+				ElementType:         types.StringType,
+				MarkdownDescription: "The start and end IP of the mount endpoint range. Expressed as a array with two entries.",
+				Computed:            true,
+			}),
```
Comment (skirsten) [inline internal/provider/filesystem_resource.go:115]: I believe this is missing the PlanModifiers

**Comment by diogopms** [comment]
@skirsten , I've updated the code based on your feedback. Can you take a look? Thanks 🙏🏻

**Review by skirsten — APPROVED** [review]

**Pull request merged**